pkg/testsuite/mock/wallet_transactions.go

```diff
@@ -321,7 +321,7 @@ func (w *Wallet) CreateBasicOutputsEquallyFromInputs(transactionName string, out
 	return signedTransaction
 }
 
-func (w *Wallet) createSignedTransactionWithOptions(transactionName string, opts ...options.Option[builder.TransactionBuilder]) (*iotago.SignedTransaction, error) {
+func (w *Wallet) CreateSignedTransactionWithOptions(transactionName string, opts ...options.Option[builder.TransactionBuilder]) (*iotago.SignedTransaction, error) {
```

Comment:
I would prefer to keep this as a private method and use utility functions for specific types of transactions. This just helps keep the tests themselves a bit cleaner and clearer.